### PR TITLE
Fixed typo in comment and minor code improvements

### PR DESCRIPTION
**Pull request by jwillemsen**
* dds/DCPS/Service_Participant.cpp:
    * examples/DCPS/Messenger_IOGR_Imr/Aggregator.cpp:
    * examples/DCPS/Messenger_IOGR_Imr/publisher.cpp:
    * performance-tests/DCPS/InfoRepo_population/subscriber.cpp:
    * performance-tests/DCPS/MulticastListenerTest/publisher.cpp:
    * performance-tests/DCPS/MulticastListenerTest/subscriber.cpp:
    * performance-tests/DCPS/SimpleE2ETest/publisher.cpp:
    * performance-tests/DCPS/SimpleE2ETest/subscriber.cpp:
    * performance-tests/DCPS/TCPListenerTest/publisher.cpp:
    * performance-tests/DCPS/TCPListenerTest/subscriber.cpp:
    * performance-tests/DCPS/TCPProfilingTest/publisher.cpp:
    * performance-tests/DCPS/TCPProfilingTest/subscriber.cpp:
    * performance-tests/DCPS/UDPListenerTest/publisher.cpp:
    * performance-tests/DCPS/UDPListenerTest/subscriber.cpp:
    * performance-tests/DCPS/UDPNoKeyTest/publisher.cpp:
    * performance-tests/DCPS/UDPNoKeyTest/subscriber.cpp:
    * tests/DCPS/BuiltInTopicTest/monitor.cpp:
    * tests/DCPS/BuiltInTopicTest/publisher.cpp:
    * tests/DCPS/BuiltInTopicTest/subscriber.cpp:
    * tests/DCPS/CompatibilityTest/publisher.cpp:
    * tests/DCPS/DCPSInfoRepo/pubsub.cpp:
    * tests/DCPS/DcpsIntegration/infrastructure.cpp:
    * tests/DCPS/DcpsIntegration/topic.cpp:
    * tests/DCPS/FooTest4/main.cpp:
    * tests/DCPS/FooTest4_0/main.cpp:
    * tests/DCPS/FooTest5/publisher.cpp:
    * tests/DCPS/FooTest5/subscriber.cpp:
    * tests/DCPS/FooTest5_0/main.cpp:
    * tests/DCPS/LivelinessTest/publisher.cpp:
    * tests/DCPS/LivelinessTest/subscriber.cpp:
    * tests/DCPS/LivelinessTimeout/publisher.cpp:
    * tests/DCPS/LivelinessTimeout/subscriber.cpp:
    * tests/DCPS/ManyTopicMultiProcess/publisher.cpp:
    * tests/DCPS/ManyTopicMultiProcess/subscriber.cpp:
    * tests/DCPS/ManyTopicTest/publisher.cpp:
    * tests/DCPS/ManyTopicTest/subscriber.cpp:
    * tests/DCPS/Monitor/publisher.cpp:
    * tests/DCPS/MultiDPTest/publisher.cpp:
    * tests/DCPS/MultiDPTest/subscriber.cpp:
    * tests/DCPS/NotifyTest/publisher.cpp:
    * tests/DCPS/NotifyTest/subscriber.cpp:
    * tests/DCPS/Reconnect/publisher.cpp:
    * tests/DCPS/Reconnect/subscriber.cpp:
    * tests/DCPS/SimpleFooTest/main.cpp:
    * tests/DCPS/ViewState/main.cpp:
    * tests/DCPS/ZeroCopyRead/main.cpp:
    * tests/DCPS/sub_init_loop/publisher.cpp:
    * tests/DCPS/sub_init_loop/subscriber.cpp:
    * tools/dcpsinfo_dump/dump.cpp:

### CODE REVIEW

File: dds/DCPS/DomainParticipantFactoryImpl.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ DomainParticipantFactoryImpl::create_participant(
     return DDS::DomainParticipant::_nil();
   }
 
-  DomainParticipantImpl* dp;
+  DomainParticipantImpl* dp = 0;
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: I'm not sure why this is needed, ACE_NEW_RETURN always assigns a value to the pointer.

File: dds/DCPS/DomainParticipantFactoryImpl.cpp [hunk 2]
```diff
@@ -100,7 +100,7 @@ DomainParticipantFactoryImpl::create_participant(
   // needed for the built in topics during enable.
   dp->set_object_reference(dp_obj);  //xxx no change
 
-  if (qos_.entity_factory.autoenable_created_entities == 1) {
+  if (qos_.entity_factory.autoenable_created_entities == true) {
```
Comment (mitza-oci) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: Remove the "== true", it doesn't add anything.

Comment (mitza-oci) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: Also I'm not in favor of "== false" when we have the simple and succinct ! operator

Comment (jwillemsen) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: In a lot of existing places OpenDDS uses ==true and ==false, I just tried to keep the code consistent. If you want to remove ==false/==true I am in favor but that should be a general cleanup action

File: dds/DCPS/DomainParticipantFactoryImpl.cpp (hunk 1 above)
Comment (jwillemsen) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: Always safer in case of refactoring of code, this way pointers are always zero

File: dds/DCPS/DomainParticipantFactoryImpl.cpp (hunk 2 above)
Comment (jwillemsen) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: That will be at least 300 places where another set of code changes are necessary, let us do that in another PR

Comment (mitza-oci) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: Your comments here seem to be based on an assumption that I don't think holds.  The code base doesn't currently have one consistent style.  It would be nice (especially if it was a *good* style), but that's not the reality for a project that was developed by a distributed team over 12+ years.

Due to this (and other reasons), there is no requirement that all source code in the project follow one style, however there is a requirement (or at least strong desire) that new changes going in to the repo are as readable as possible and follow the recommended style.

There are always trade-offs here, no solution will be perfect.  But what we've been doing with OpenDDS recently is to make changes incrementally as we're updating sections of the code.  I don't want to have to think about both the "good" style (current) and also keep in mind a bunch of old styles too.  This also avoids huge commits across all source files that can be accidentally destabilizing and also clutter the history with irrelevant changes.

Comment (jwillemsen) [inline dds/DCPS/DomainParticipantFactoryImpl.cpp]: I am 100% in favor of improving the code in general, looks with this change I focused to much on consistent style instead of general improvements. I will fix the comparison of autoenable_created_entities right now, give me a few minutes

**Comment by jwillemsen** [comment]
Pushed some changes
